pkg/lib/fieldgroups/database/database_validator.go

```diff
@@ -28,8 +31,38 @@ func (fg *DatabaseFieldGroup) Validate(opts shared.Options) []shared.ValidationE
 		ca = fg.DbConnectionArgs.Ssl.Ca
 	}
 
+	sslrootcertTmpPath := fg.DbConnectionArgs.SslRootCert
+	if fg.DbConnectionArgs.SslMode == "verify-full" || fg.DbConnectionArgs.SslMode == "verify-ca" {
+		// Write database.pem needed for db validation, if any, to a temp file
+		tmpCert, err := ioutil.TempFile("/tmp", "database.*.pem")
+		if err != nil {
+			newError := shared.ValidationError{
+				Tags:       []string{"DB_URI"},
+				FieldGroup: fgName,
+				Message:    "Could not connect to database. Error: " + err.Error(),
+			}
+			errors = append(errors, newError)
+			return errors
+		}
+
+		defer os.Remove(tmpCert.Name())
```

Comment:
This will leak a file descriptor. You need to close the file before removing it otherwise the fd will still be allocated.

Replace this by what follows and remove what is in line 51 below:

```go
defer func() {
    tmpCert.Close()
    os.Remove(tmpCert.Name())
}()
```

Reply:
Done